### PR TITLE
Bump various test dependencies

### PR DESCRIPTION
**Pull request by AlexWaygood**

### CODE REVIEW

**Comment by srittau** [comment]
Could we put the black reformats into a separate PR? Remember that the commit message becomes a changelog entry for third-party stubs.

**Comment by AlexWaygood** [comment]
> Could we put the black reformats into a separate PR? Remember that the commit message becomes a changelog entry for third-party stubs. (Something I always forget when merging.)

Yes, makes sense

File: stdlib/builtins.pyi [hunk 1]
```diff
@@ -1852,6 +1852,7 @@ class BaseException:
     __context__: BaseException | None
     __suppress_context__: bool
     __traceback__: TracebackType | None
+    def __new__(cls: type[Self], *args: object) -> Self: ...
```
Comment (AlexWaygood) [inline stdlib/builtins.pyi:1855]:
```suggestion
```

**Comment by AlexWaygood** [comment]
The third-party stubtest failure in `beautifulsoup` has already been noted in #9645, and is unrelated to this PR.

**Review by JelleZijlstra — APPROVED** [review]

**Pull request merged**